src/Morpher.php

```diff
@@ -31,7 +44,10 @@ protected function prepareMorphs($event)
             return;
         }
 
-        $this->getMorphs($event->migration)->each(fn($morphs) => $morphs->prepare());
+        $this->getMorphs($event->migration)
+            ->each(fn($morph) => $morph->withConsole($this->console))
+            ->each(fn($morph) => $morph->prepare())
+            ->each(fn($morph) => $this->inspectionsForMorph($morph)->each->runBeforeMigrating($morph));
```

Comment:
I think these `beforeMigrating` hooks need to run before `prepare` to gain access to the current db state in the Morph `prepare` methods.

Reply:
Good spot. Changed